CMakeLists.txt

```diff
@@ -368,13 +370,17 @@ if(USE_PTHREADS)
 	endif()
 endif()
 
-if(HAVE_NEWLOCALE AND HAVE_FREELOCALE AND HAVE_USELOCALE OR APPLE)
+if(HAVE_NEWLOCALE AND HAVE_FREELOCALE OR APPLE)
 	list(APPEND FUNCTION_DEFINES HAVE_LOCALE)
 	if (HAVE_SNPRINTF_L)
 		list(APPEND FUNCTION_DEFINES HAVE_SNPRINTF_L)
 	endif()
 endif()
 
+if(HAVE_USELOCALE OR APPLE)
+	list(APPEND FUNCTION_DEFINES HAVE_USELOCALE)
+endif()
+
```

Comment:
These lines are unnecessary--`HAVE_USELOCALE` will be defined within `find_function`.